simpleclient_spring_boot/src/main/java/io/prometheus/client/spring/EnablePrometheusCollectorRegistration.java

```diff
@@ -0,0 +1,52 @@
+package io.prometheus.client.spring;
+
+import org.springframework.context.annotation.Import;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * An {@literal @}Enable style annotation that provides CollectorRegistry related functions.
+ *
+ * <pre><code>
+ *{@literal @}Configuration
+ *{@literal @}EnablePrometheusCollectorRegistration(useDefault = true, register = true)
+ * public class MyConfiguration {
+ *
+ *  {@literal @}Bean
+ *   public Counter counter() {
```

Comment:
> Static field singletons are somewhat notorious for causing memory leaks in containers with multiple classloaders, especially when they're used as a registry.

In this context that's not a memory leak, that's correct behaviour. The Collector should continue to exist and be registered until the process shutdowns.

> Here's a good write up about singletons & safety

I don't see any indication there that the existing client_java code is incorrect.